handler/src/main/java/io/netty/handler/ssl/ReferenceCountedOpenSslEngine.java

```diff
@@ -237,36 +237,38 @@ protected void deallocate() {
         keyMaterialManager = context.keyMaterialManager();
         enableOcsp = context.enableOcsp;
 
-        Lock readerLock = context.ctxLock.readLock();
-        readerLock.lock();
-        try {
-            ssl = SSL.newSSL(context.ctx, !context.isClient());
-        } finally {
-            readerLock.unlock();
-        }
-        try {
-            networkBIO = SSL.bioNewByteBuffer(ssl, context.getBioNonApplicationBufferSize());
+        synchronized (this) {
+            Lock readerLock = context.ctxLock.readLock();
+            readerLock.lock();
+            try {
+                ssl = SSL.newSSL(context.ctx, !context.isClient());
```

Comment:
@Scottmitch is it shared between ELGs?  If there were multiple separate ELGs, what would happen?